optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/common/accessor/gizmo/GizmoMemberAccessor.java

```diff
@@ -0,0 +1,637 @@
+import org.jboss.jandex.IndexView;
+import org.jboss.jandex.MethodInfo;
+import org.optaplanner.core.impl.domain.common.ReflectionHelper;
+import org.optaplanner.core.impl.domain.common.accessor.MemberAccessor;
+
+import io.quarkus.gizmo.ClassCreator;
+import io.quarkus.gizmo.ClassOutput;
+import io.quarkus.gizmo.FieldDescriptor;
+import io.quarkus.gizmo.MethodCreator;
+import io.quarkus.gizmo.MethodDescriptor;
+import io.quarkus.gizmo.ResultHandle;
+
+public class GizmoMemberAccessor {
+
+    /**
+     * Stores the generic type of a member (required
+     * as we cannot hard code a non-primitive object
+     * instance in Gizmo code); will be accessed
+     * from generated Gizmo code via getGenericTypeFor
+     */
+    private static Map<String, Type> gizmoMemberAccessorNameToGenericType = new HashMap<>();
+
+    /**
+     * Stores the annotated element of a member (required
+     * as we cannot hard code a non-primitive object
+     * instance in Gizmo code); will be accessed
+     * from generated Gizmo code via getAnnotatedElementFor
+     */
+    private static Map<String, AnnotatedElement> gizmoMemberAccessorNameToAnnotatedElement = new HashMap<>();
+
+    /**
+     * The Gizmo generated bytecode. Used by
+     * gizmoClassLoader when not run in Quarkus
+     * in order to create an instance of the Member
+     * Accessor
+     */
+    private static final Map<String, byte[]> classNameToBytecode = new HashMap<>();
+
+    /**
+     * A custom classloader that looks for the class in
+     * classNameToBytecode
+     */
+    private static ClassLoader gizmoClassLoader = new ClassLoader() {
+        public String getName() {
+            return "";
+        }
+
+        @Override
+        public Class<?> findClass(String name) throws ClassNotFoundException {
+            if (classNameToBytecode.containsKey(name)) {
+                // Gizmo generated class
+                byte[] byteCode = classNameToBytecode.get(name);
+                return defineClass(name, byteCode, 0, byteCode.length);
+            } else {
+                // Not a Gizmo generated class; load from context class loader
+                return Thread.currentThread().getContextClassLoader().loadClass(name);
+            }
+        }
+    };
+
+    private static boolean useClassLoader = false;
+
+    /**
+     * Call this method if you generated the bytecode
+     * during the build process (ex: Quarkus)
+     *
+     * @param classLoader The classloader that contains the generated
+     *        gizmo classes.
+     *
+     * @param newGizmoMemberAccessorNameToGenericType The map from
+     *        member accessor class name
+     *        to its generic type
+     * @param newGizmoMemberAccessorNameToAnnotatedElement The map from
+     *        member accessor class name
+     *        to its annotated Element.
+     */
+    public static void setGizmoClassLoader(ClassLoader classLoader, Map<String, Type> newGizmoMemberAccessorNameToGenericType,
```

Comment:
Thinking
What about renaming the method to
`usePregeneratedBytecodeAndMaps`
and changing `useClassloader` to `usePregeneratedBytecodeFromClassLoader`?

Reply:
Sounds much better!

A related question: when do we take these maps from outside Quarkus?

Reply:
When the OptaPlanner ScoreFactory bean get initialized from Quarkus. (Outside Quarkus, this would only be called if the bytecode was already generated (say via a build time processor or equivalent).